test/built-ins/Temporal/Calendar/prototype/monthDayFromFields/overflow-reject.js

```diff
@@ -38,27 +42,21 @@ const cal = new Temporal.Calendar("iso8601");
   );
 });
 
-assert.throws(RangeError, () => cal.monthDayFromFields(
-    { monthCode: "M01", day: 32 }, { overflow: "reject" }), "Day 32 is out of range for monthCode M01");
-assert.throws(RangeError, () => cal.monthDayFromFields(
-    { monthCode: "M02", day: 30 }, { overflow: "reject" }), "Day 30 is out of range for monthCode M02");
-assert.throws(RangeError, () => cal.monthDayFromFields(
-    { monthCode: "M03", day: 32 }, { overflow: "reject" }), "Day 32 is out of range for monthCode M03");
-assert.throws(RangeError, () => cal.monthDayFromFields(
-    { monthCode: "M04", day: 31 }, { overflow: "reject" }), "Day 31 is out of range for monthCode M04");
-assert.throws(RangeError, () => cal.monthDayFromFields(
-    { monthCode: "M05", day: 32 }, { overflow: "reject" }), "Day 32 is out of range for monthCode M05");
-assert.throws(RangeError, () => cal.monthDayFromFields(
-    { monthCode: "M06", day: 31 }, { overflow: "reject" }), "Day 31 is out of range for monthCode M06");
-assert.throws(RangeError, () => cal.monthDayFromFields(
-    { monthCode: "M07", day: 32 }, { overflow: "reject" }), "Day 32 is out of range for monthCode M07");
-assert.throws(RangeError, () => cal.monthDayFromFields(
-    { monthCode: "M08", day: 32 }, { overflow: "reject" }), "Day 32 is out of range for monthCode M08");
-assert.throws(RangeError, () => cal.monthDayFromFields(
-    { monthCode: "M09", day: 31 }, { overflow: "reject" }), "Day 31 is out of range for monthCode M09");
-assert.throws(RangeError, () => cal.monthDayFromFields(
-    { monthCode: "M10", day: 32 }, { overflow: "reject" }), "Day 32 is out of range for monthCode M10");
-assert.throws(RangeError, () => cal.monthDayFromFields(
-    { monthCode: "M11", day: 31 }, { overflow: "reject" }), "Day 31 is out of range for monthCode M11");
-assert.throws(RangeError, () => cal.monthDayFromFields(
-    { monthCode: "M12", day: 32 }, { overflow: "reject" }), "Day 32 is out of range for monthCode M12");
+TemporalHelpers.ISOMonths.forEach(({ month, monthCode, daysInMonth }) => {
```

Comment:
This consolidates the above and covers the smallest affected day for each month.